test/test_cuda.py

```diff
@@ -4702,6 +4704,84 @@ class TestCudaOptims(TestCase):
     # These tests will be instantiate with instantiate_device_type_tests
     # to apply the new OptimizerInfo structure.
 
+    @onlyNativeDeviceTypes
+    @optims(
+        [optim for optim in optim_db if "fused" in optim.supported_impls],
+        dtypes=[torch.float32],
+    )
+    def test_grad_scaling_autocast_fused_optimizers(self, device, dtype, optim_info):
+        if device not in optim_info.supports_fused_on:
```

Comment:
```suggestion
        if device.type not in optim_info.supports_fused_on:
```
AdamW, Adam and SGD should not get skipped for "cuda:0" not being in the list. This should really use device.type.

Reply:
since device is coming as str, doing ```device = device.split(":")[0]```